src/dmtcp_coordinator.cpp

```diff
@@ -467,6 +468,7 @@ void DmtcpCoordinator::updateMinimumState(WorkerState oldState)
     if (exitAfterCkpt || exitAfterCkptOnce) {
       JNOTE("Checkpoint Done. Killing all peers.");
       broadcastMessage(DMT_KILL_PEER);
+      JTIMER_STOP ( checkpoint );
```

Comment:
I tend to agree with @karya0.  @rohgarg, the previous case that you mention stops the timer after finishing the resume barrier.  That makes sense to me.  The cost of a checkpoint should include the cost of the corresponding "resume".

However, the cost of a "checkpoint" during `--exit-on-checkpoint` should be just the cost of the `checkpoint`.  It should not include the cost of the `exit` (the cost of a barrier to tell all peers to exit).